StatefulReflectiveBoundTexture.go

```diff
@@ -0,0 +1,180 @@
+package giu
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrNeedReset is an error indicating that the surface cannot be loaded without a reset.
+// The method (*StatefulReflectiveBoundTexture).ResetState() should be called.
+var ErrNeedReset = errors.New("cannot load surface without a reset. Should call (*StatefulReflectiveBoundTexture).ResetState()")
+
+// ErrIsLoading is an error indicating that the surface state cannot be reset while loading.
+var ErrIsLoading = errors.New("cannot reset surface state while loading")
+
+// SurfaceState represents the state of the surface.
+type SurfaceState int
+
+//go:generate stringer -type=SurfaceState
+const (
+	// SsNone indicates that the surface state is none.
+	SsNone SurfaceState = iota
+	// SsLoading indicates that the surface is currently loading.
+	SsLoading
+	// SsFailure indicates that the surface loading has failed.
+	SsFailure
+	// SsSuccess indicates that the surface loading was successful.
+	SsSuccess
+)
```

Comment:
Could you replace `Ss` with the full name? giu has many constants and would be easier to identify them with the full name.